### ISSUE TITLE
ESLint throwing syntax error on valid arrow function?

### ISSUE THREAD

**Issue by Nolski** [eslint/eslint#4283]
Hi,
I am receiving a syntax error when passing `window.setTimeout(() => { console.log("foo"); }, 1)`. The syntax error is being thrown at the arrow function `>`. I believe I am initializing ESLint correctly with ES6 settings.

```javascript
let eslint = new ESLint.CLIEngine({
  ignore: false,
  rulePaths: ['dist/eslint'],
  rules: ESLintRules,
  useEslintrc: false,
  es6: true,
  ecmaFeatures: {
    arrowFunctions: true,
  }
});
```

This is running eslint `1.7.2`

Output:
```javascript
{ 
    code: 'JS_SYNTAX_ERROR',
    column: 23,
    description: 'There is a JavaScript syntax error in your code; validation cannot continue on this file.',
    file: 'badcode.js',
    line: 1,
    message: 'JavaScript syntax error',
    sourceCode: undefined,
    type: 'error' 
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Works fine on online demo with `1.7.3` here: http://eslint.org/demo/

**Comment by nzakas** [comment]
You're providing the wrong information to `CLIEngine`. There's no `es6` or `ecmaFeatures` options. Please see http://eslint.org/docs/developer-guide/nodejs-api.html#cliengine for the full set of options available.
